bftengine/src/bftengine/ControlStateManager.cpp

```diff
@@ -159,21 +110,8 @@ bool ControlStateManager::verifyUnwedgeSignatures(
   for (auto const& sig : signatures) {
     std::string sig_data = std::to_string(sig.first) + std::to_string(last_checkpoint.value());
     std::string signature(sig.second.begin(), sig.second.end());
-    bool valid = false;
-    if (sig.first == ReplicaConfig::instance().replicaId) {
-      // SigManager cannot verify signature coming from same replica
-      // Generate the signature again and compare them.
-      verified_sigs++;
-      std::string sign(sig_manager->getMySigLength(), '\0');
-      sig_manager->sign(sig_data.c_str(), sig_data.size(), sign.data(), sign.size());
-      if (sign == std::string(sig.second.begin(), sig.second.end())) {
-        valid = true;
-        verified_sigs++;
-      }
```

Comment:
Is this issue fixed in the SigManager?
If not, you will have reported one less valid signature by removing this block

Reply:
yes, I fixed it a few weeks ago

Reply:
Great! Thanks a lot!